uaclient/util.py

```diff
@@ -835,8 +842,18 @@ def should_reboot(installed_pkgs: Optional[Set[str]] = None) -> bool:
         # If the file doesn't exist, we will default to the
         # reboot  marker file
         return True
-    else:
-        return len(installed_pkgs.intersection(reboot_required_pkgs)) != 0
+
+    if installed_pkgs is not None:
+        if len(installed_pkgs.intersection(reboot_required_pkgs)) != 0:
+            return True
+
+    if installed_pkgs_regex is not None:
+        for pkg_name in reboot_required_pkgs:
+            for pkg_regex in installed_pkgs_regex:
+                if re.search(pkg_regex, pkg_name):
+                    return True
```

Comment:
I am assuming we need this bit here because we are installing a metapackage that will not appear on the `reboot_required_pkgs` file right ?

Reply:
Yep we apt install linux-realtime but the package that adds itself to that file is something like linux-$VERSION-realtime